### ISSUE TITLE
Extract product-shaped content into profiles and add an original JSON Demo

### ISSUE THREAD

**Issue by Extender92** [Extender92/Monopoly#4]
## Problem/Goal

Remove bundled regional and product-shaped board/card content, replace it with generic identifiers, and provide a fully original JSON-based Demo profile for the public engine.

## Required behavior

- Use ProfileId, ProfileRevision, ProfileFingerprint, SpaceId, DeckId, and CardId independently of text and collection order.
- Move board, presentation, setup, deck, card, policy, and effect data into validated JSON profiles.
- Replace regional card classes, named destinations, fixed deck roles, and product-shaped factories with generic capabilities and declarative effects from #72–#74.
- Provide one redistributable Demo profile with its own fictional theme, wording, economy, ordering, topology, values, group sizes, and event set.
- Demo Version 1 declares no finite bank inventory because the supported baseline contains no inventory capability.
- Demo has a non-40 track, no four equivalent special corners, one event deck, and a profile-defined round-limited score win.
- Bundle no official profile, compatibility pack, official wording, or renamed one-to-one product dataset.
- Use small structurally distinct synthetic profiles for focused tests.
- Reject legacy Version 1 saves clearly after this transition; accept a development-only persistence gap until #52, with no release during the gap.
- Accept a development-only match-creation gap after removing legacy content: Console reports new/load as temporarily unavailable until #40 and #75 provide validated-profile setup and execution.

## Scope

JSON Demo data, removal of legacy presentation content, generic IDs/effects, test fixtures, the compile-safe WIP transition, and documentation needed to make the default tree self-contained.

## Out of scope

A universal board-game runtime, profile editor, arbitrary scripts, private profile data, Save V2, generic match setup, capability execution, licensing approval, clean publication, and legacy-history rewriting.

## Acceptance criteria

- [x] No public Core enum, card type, destination, factory, test, or default data path selects a regional or official edition.
- [x] The bundled Demo validates through #74 and uses the contracts in #72 and #73.
- [x] Demo has project-owned names, texts, layout, economy, identifiers, event data, and score-win policy and is not a relabeled official dataset.
- [x] Demo Version 1 has no finite inventory state.
- [x] Focused tests use small synthetic profiles with varied track lengths and deck counts.
- [x] No official profile or compatibility package is tracked or emitted.
- [x] Duplicate IDs, broken targets, unsupported capabilities, and invalid ranges fail before match creation.
- [x] Version 1 saves fail with a typed compatibility result once the transition lands, and save attempts write nothing until #52.
- [x] Console remains buildable and reports the temporary new/load gap until #40 and #75 complete the runtime path.
- [x] #55 source and artifact rules classify the replacement and pass with no unknown findings.
- [x] Release build and tests pass without new warnings and without an external profile.

## Dependencies

- Blocked by #71, #72, #73, and #74.
- Coordinates presentation metadata with #32 and setup with #40.
- #40 restores validated-profile match creation, #75 replaces the legacy executor, and #52 restores persistence after this transition.
- #56 applies neutral code identity after the profile boundary is stable.

## Related issues

- Checklist child of #37 and #59.
- Related to #54, #55, #57, #58, and #78.

**Comment by Extender92** [comment]
## Rule-profile and regional data requirements

The UK/US review for #37 adds concrete scope to this data refactor:

- Board and card creation should consume a validated rule profile rather than branching only on `GameRules.Language`.
- UK Classic, US Classic and Custom should use the same builders and domain types.
- Regional data must use stable identifiers so display names can differ without changing save references or game flow.
- Verify every property, price, rent, mortgage value, building cost, tax, Station/Railroad, Utility and card against the exact documented Hasbro baseline edition.
- Correct known regional leakage such as the UK board currently receiving the US `Luxury Tax` name instead of UK `Super Tax`.
- Keep currency, regional names and card text in profile data; keep movement/payment effects in reusable Core behavior.
- Validate board size, unique positions, group membership, card counts/identities and required special squares when constructing a profile.
- Custom profiles should derive from an explicit base and apply validated overrides without copying the complete production data set.
- Add profile-level data verification tests for both official editions and representative Custom overrides.
- Update `docs/game-rules.md` after the implementation so the data baseline is documented without referring to issue numbers.

Related to #27, #32 and #37.

**Comment by Extender92** [comment]
## Concrete findings from the Classic rule audit

The documentation audit against the selected UK C1009 (2021) and US C1009 (2017) baselines found current data mismatches that belong in this refactor:

- UK currently uses the US `Luxury Tax` name at position 38; the UK profile requires `Super Tax`.
- UK Chance currently contains 15 entries while the selected Classic profile requires 16 card identities; duplicate effect cards must still have distinct stable identities.
- Current cards are identified by queue index/text rather than stable regional card IDs.
- Current `Language` branching combines regional identity, currency and data selection instead of resolving a validated profile.
- Player starting balance is stored on `Player` rather than supplied by profile setup; implementation is tracked in #40.
- Regional data tests check only a few positions/values and do not verify complete 40-space boards, 28 deeds, 16+16 cards or uniqueness.

Keep exact long card wording in sourced profile data rather than duplicating it in documentation. Tests should verify identity, effect mapping, deck size and required numeric values for each selected publication.

Related to #37, #40 and #42.
